src/game.cpp

```diff
@@ -10723,9 +10726,33 @@ void game::place_player( const tripoint &dest_loc )
     // Important: don't use dest_loc after this line. `update_map` may have shifted the map
     // and dest_loc was not adjusted and therefore is still in the un-shifted system and probably wrong.
 
-    //Auto pulp or butcher
-    if( get_option<bool>( "AUTO_PULP_BUTCHER" ) && mostseen == 0 ) {
-        const std::string pulp_butcher = get_option<std::string>( "AUTO_PULP_BUTCHER_ACTION" );
+    //Auto pulp or butcher and Auto foraging
+    if( get_option<bool>( "AUTO_FEATURES" ) && mostseen == 0 ) {
+        static const direction adjacentDir[8] = { NORTH, NORTHEAST, EAST, SOUTHEAST, SOUTH, SOUTHWEST, WEST, NORTHWEST };
+
+        const std::string forage_type = get_option<std::string>( "AUTO_FORAGING" );
+        if( forage_type != "off" ) {
```

Comment:
A little bit late, but I guess we also need to add `!forage_type.empty() && ` here.

Reply:
Maybe you are right.

I just run into some issues with `forage_type` which was set to NULL and caused crash several lines later while testing another unrelated PR.

Reply:
See - appears after I debug teleport to some place:

![image](https://user-images.githubusercontent.com/16213433/46883138-b3dfcd00-ce59-11e8-83bc-6fc1fac31d80.png)